File: 1. 단계별로_풀어보기/10~19/18. 심화 2/2. 25192.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

usedUser = set()
result = 0

for _ in range(N):
  log = input().strip()

  if log == 'ENTER':
    usedUser = set()
  else:
    if log not in usedUser:
      usedUser.add(log)
      result += 1


output(f"{result}")
  

File: 1. 단계별로_풀어보기/10~19/18. 심화 2/2. 25192.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const logArr = inputArr.slice(1).map(row => row.trim())

// 곰곰티콘 보낸 유저를 Set 형태로 저장
let dict = {}

// 곰곰티콘 사용횟수
let result = 0

// 로그마다 연산
logArr.forEach(log => {
  if (log === 'ENTER') {
    dict = {}
    return
  }

  if (dict[log] !== 1) {
    dict[log] = 1
    result += 1
  }
})

// 결과 출력
process.stdout.write(`${result}`)
